File: easy/panacea.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    my $r = 0;
    my ( $ls, $rs ) = split( / \| /, $line );
    map { $r += hex $_ } split( / /, $ls );
    map { $r -= oct( "0b" . $_ ) } split( / /, $rs );
    print( $r > 0 ? "False\n" : "True\n" );
}
close(INFILE);

File: easy/panacea.rb
# sum
class Array
  def sum
    inject(0, :+)
  end
end

File.open(ARGV[0]).each_line do |line|
  s = line.split(' | ').map(&:split)
  if s[0].map { |x| x.to_i(16) }.sum > s[1].map { |x| x.to_i(2) }.sum
    puts 'False'
  else puts 'True'
  end
end
